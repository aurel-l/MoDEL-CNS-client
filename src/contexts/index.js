import { createContext } from 'react';

export const AccessionCtx = createContext(null);
export const ProjectCtx = createContext(null);
export const PdbCtx = createContext(null);
