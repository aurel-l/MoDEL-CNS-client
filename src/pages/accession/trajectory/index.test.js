import React from 'react';
import { shallow } from 'enzyme';

import Trajectory, { TrajectoryMetadata } from '.';
import { ProjectCtx } from '../../../contexts';

describe('<Trajectory />', () => {
  it('should render', () => expect(shallow(<Trajectory />)).toMatchSnapshot());
});

describe('<TrajectoryMetadata />', () => {
  const metadata = {
    BOXSIZEX: 13.66313,
    BOXSIZEY: 13.66313,
    BOXSIZEZ: 14.49403,
    BOXTYPE: 'Triclinic',
    CL: 263,
    DPPC: 601,
    ENSEMBLE: 'NPT',
    FF: 'gromos53a6',
    FREQUENCY: 100,
    LENGTH: 100,
    MEMBRANE: 'DPPC',
    NA: 251,
    PCOUPLING: 'Semiisotropic',
    PROT: 570,
    PROTATS: 6050,
    SNAPSHOTS: 1000,
    SOL: 61651,
    SYSTATS: 221567,
    TEMP: 300,
    TIMESTEP: 2,
    WAT: 'TIP3P',
    atomCount: 36100,
    frameCount: 10001,
  };
  it('should render', () =>
    expect(
      shallow(
        <ProjectCtx.Provider value={{ metadata }}>
          <TrajectoryMetadata />
        </ProjectCtx.Provider>,
      ),
    ).toMatchSnapshot());
});
