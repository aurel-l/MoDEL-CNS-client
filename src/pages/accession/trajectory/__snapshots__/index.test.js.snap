// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Trajectory /> should render 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Card))
    className="card"
  >
    <WithStyles(ForwardRef(CardContent))>
      <Memo() />
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(Card))
    className="card"
  >
    <ForwardRef
      className="container"
    />
  </WithStyles(ForwardRef(Card))>
</Fragment>
`;

exports[`<TrajectoryMetadata /> should render 1`] = `<Memo() />`;
